src/libsystemd/sd-journal/journal-file.c

```diff
@@ -2078,6 +2078,19 @@ static int entry_item_cmp(const EntryItem *a, const EntryItem *b) {
         return CMP(le64toh(a->object_offset), le64toh(b->object_offset));
 }
 
+static size_t remove_duplicate_entry_items(EntryItem *items, size_t n) {
+        size_t j = 1;
+
+        if (n == 0)
+                return 0;
```

Comment:
doesn't matter, but you could write this as:

```c
if (n <= 1)
        return n;
```

i.e. zero length and single item arrays never have duplicates...